File: server/src/utils/secrets.ts
import { configDotenv } from 'dotenv';

configDotenv({
  path: '.env',
});

const secrets = {
  database_url: process.env.DATABASE_URL,

  project_id: process.env.FIREBASE_PROJECT_ID,
  private_key_id: process.env.FIREBASE_PRIVATE_KEY_ID,
  private_key: process.env.FIREBASE_PRIVATE_KEY,
  client_email: process.env.FIREBASE_CLIENT_EMAIL,
  client_id: process.env.FIREBASE_CLIENT_ID,
  auth_uri: process.env.FIREBASE_AUTH_URI,
  token_uri: process.env.FIREBASE_TOKEN_URI,
  auth_provider_x509_cert_url: process.env.FIREBASE_AUTH_CERT_URL,
  client_x509_cert_url: process.env.FIREBASE_CLIENT_CERT_URL,

  aws_access_key: process.env.AWS_ACCESS_KEY,
  aws_secret: process.env.AWS_SECRET,

  youtube_api_key: process.env.YOUTUBE_API_KEY,
};

export default secrets;

File: server/lib/secrets.go
package lib

import (
	"fmt"
	"os"

	"github.com/joho/godotenv"
)

type Secrets struct {
	DatabaseURL             string
	RailwayURL              string
	DockerDatabase          string
	RdsDatabase             string
	ProjectId               string
	PrivateKeyId            string
	PrivateKey              string
	ClientEmail             string
	ClientId                string
	AuthUri                 string
	TokenUri                string
	AuthProviderX509CertUrl string
	ClientX509CertUrl       string
	AwsAccessKey            string
	AwsSecret               string
	YoutubeApiKey           string
	StripeSecretKey         string
	StripeSecretKeyTest     string
	EmailHost               string
	EmailPass               string
	EmailUser               string
	EmailPort               string
}

func NewSecrets() (*Secrets, error) {

	env := os.Getenv("ENV")
	if env != "production" {
		if err := godotenv.Load(".env"); err != nil {
			return nil, fmt.Errorf("failed to load configurations file: %w", err)
		}
	}

	return &Secrets{
		// Database
		DatabaseURL:    os.Getenv("RDS_DATABASE"),
		RailwayURL:     os.Getenv("RAILWAY_URL"),
		DockerDatabase: os.Getenv("DOCKER_DATABASE"),
		RdsDatabase:    os.Getenv("RDS_DATABASE"),
		// Firebase
		ProjectId:               os.Getenv("FIREBASE_PROJECT_ID"),
		PrivateKeyId:            os.Getenv("FIREBASE_PRIVATE_KEY_ID"),
		PrivateKey:              os.Getenv("FIREBASE_PRIVATE_KEY"),
		ClientEmail:             os.Getenv("FIREBASE_CLIENT_EMAIL"),
		ClientId:                os.Getenv("FIREBASE_CLIENT_ID"),
		AuthUri:                 os.Getenv("FIREBASE_AUTH_URI"),
		TokenUri:                os.Getenv("FIREBASE_TOKEN_URI"),
		AuthProviderX509CertUrl: os.Getenv("FIREBASE_AUTH_PROVIDER_X509_CERT_URL"),
		ClientX509CertUrl:       os.Getenv("FIREBASE_CLIENT_X509_CERT_URL"),
		// AWS
		AwsAccessKey: os.Getenv("AWS_ACCESS_KEY"),
		AwsSecret:    os.Getenv("AWS_SECRET_KEY"),
		// Youtube
		YoutubeApiKey: os.Getenv("YOUTUBE_API_KEY"),
		// Stripe
		StripeSecretKey:     os.Getenv("STRIPE_SECRET_KEY"),
		StripeSecretKeyTest: os.Getenv("TEST_SECRET_KEY"),
		// Email
		EmailHost: os.Getenv("EMAIL_HOST"),
		EmailPass: os.Getenv("EMAIL_PASSWORD"),
		EmailUser: os.Getenv("EMAIL_USER"),
		EmailPort: os.Getenv("EMAIL_PORT"),
	}, nil
}
